### PR TITLE
Error reporting

### PR DESCRIPTION
**Pull request by enxebre**
This is to satisfy contract with the machine controller so it can react appropriately and signal a machine as failed with the given error message when an InvalidMachineConfiguration error is reported, main cases are:
- The spec is invalid:
	Unable to be unmarshalled
	Missing expected labels by design
	Referencing non existing kubernetes resources
- The input values provoke a client error 4xx response from the cloud call to create a machine.

openshift/enhancements#11
openshift/cluster-api#75

### CODE REVIEW

**Comment by enxebre** [comment]
/hold
openshift/cluster-api#75

File: pkg/actuators/machine/actuator.go [hunk 1]
```diff
@@ -300,34 +295,44 @@ func (a *Actuator) CreateMachine(cluster *clusterv1.Cluster, machine *machinev1.
...
-		return nil, a.handleMachineError(machine, apierrors.CreateMachine("error updating load balancers: %v", err), createEventAction)
+		return nil, a.handleMachineError(machine, err, createEventAction)
 	}
 
 	a.eventRecorder.Eventf(machine, corev1.EventTypeNormal, "Created", "Created Machine %v", machine.Name)
 	return instance, nil
 }
 
+func (a *Actuator) getUserData(machine *machinev1.Machine, machineProviderConfig *providerconfigv1.AWSMachineProviderConfig) ([]byte, error) {
+	if machineProviderConfig.UserDataSecret != nil {
```
Comment (bison) [inline pkg/actuators/machine/actuator.go]: I realize you're just moving this around (which is already an improvement) but can we make this just bail early if the secret is nil? As is, 100% of the function is wrapped in this if statement.

**Pull request merged**